rootfs/scheduler/__init__.py

```diff
@@ -993,6 +993,10 @@ def _wait_until_pods_are_ready(self, namespace, container, labels, desired):  #
             logger.info('timed out ({}s) waiting for pods to come up in namespace {}'.format(timeout, namespace))  # noqa
 
         logger.info("{} out of {} pods in namespace {} are in service".format(count, desired, namespace))  # noqa
+        if count != desired:
+            # raising to allow operations to rollback
+            raise KubeException('Not enough pods in namespace {} came into service. '
+                                '{} out of {}'.format(namespace, count, desired))
```

Comment:
That's something that has been around for a longgggg time (probably v1 days) - That's just the waiting aspect for each *scale* operation, of which can be many in a single deploy. So far `HEALTHCHECK_INITIAL_DELAY` can extend that timeout tho. So if you have a delay of 120 seconds then it is a total of 240 seconds timeout per scale operation.